ui/shared/LogoMenu.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const choices = [
+  { url: '/', text: 'Treeherder' },
+  { url: '/perf.html', text: 'Perfherder' },
+  { url: '/intermittent-failures.html', text: 'Intermittent Failures' },
+];
+
+export default function LogoMenu(props) {
+  const { menuText, menuTitle, menuImage } = props;
+
+  const menuChoices = choices.filter(choice => choice.text !== menuText);
+
+  return (
+    <span className="dropdown">
+      <button
+        id="th-logo"
+        title={menuTitle}
+        data-toggle="dropdown"
+        className="btn btn-view-nav dropdown-toggle"
+      >
+        <img src={menuImage} alt={menuText} />
+      </button>
+      <ul className="dropdown-menu" role="menu" aria-labelledby="th-logo">
+        {menuChoices.map(choice => (
+          <li key={choice.text}><a href={choice.url} className="dropdown-item">{choice.text}</a></li>
+        ))}
+      </ul>
+    </span>
+  );
+}
+
+LogoMenu.propTypes = {
+  menuText: PropTypes.string.isRequired,
+  menuTitle: PropTypes.string.isRequired,
+  menuImage: PropTypes.string,
+};
+
+LogoMenu.defaultProps = {
+  menuImage: null,
```

Comment:
Does omitting the `null` here give a prop validation error?

Reply:
It does, yeah.  Anything that's not ``.isRequired`` has to have a default, even if it's null.  I guess it's a way to be sure you're intentionally allowing that.

Reply:
The component doesn't currently handle it being null, so should we make it required until it's used by a page that doesn't set the image? Alternatively we could add handling for no image now.

(I see prop types as a guarantee as to what are valid arguments and at the moment that guarantee doesn't hold)

Reply:
So, for me, if I omit the ``menuImage`` param on the ``LogoMenu`` in ``PrimaryNavBar``, then it correctly falls back to the ``alt``/``menuText`` text.  Are you seeing something different?

Reply:
Ah good it doesn't fail per-se (I didn't test locally); though I was also thinking that the `<img>` element should be omitted entirely if there is no image URL.